docs/END-TO-END-TESTING.md

```diff
@@ -0,0 +1,103 @@
+## TestCafe
+This guide aims to share how TestCafe is being used in Plio for performing End-to-End (E2E) and integration tests.
+
+#### End-to-End (E2E) Testing
+End-to-End testing sits on top of integration testing. While integration testing is limited to check integration between two functional components, end-to-end testing is more focused on the whole application and integration of several components together in an end-to-end workflow.
+End-to-End testing generally mimics the user behavior on an actual browser and the actions they may take while using the application.
+
+#### BrowserStack
+Plio uses BrowserStack to enable automated End-to-End cross-browser and cross-OS testing.
+[BrowserStack](https://www.browserstack.com/) is a cloud-based service that provides a virtual browser for testing web application. It is a free service that allows you to test your web application on a variety of devices and browsers.
+Plio's [Continuous Integration workflow](../.github/workflows/ci.yml) runs the E2E tests on the following operating systems and browsers using BrowserStack:
+1. Chrome v96 (Windows 10, OSX Big Sur, OSX Catalina)
+2. Firefox v95 (Windows 10, OSX Big Sur, OSX Catalina)
+3. Edge v96 (Windows 10)
+4. Safari v14.1 (OSX Big Sur)
+5. Safari v13.1 (OSX Catalina)
+
+
+### Pre-requisites
+For login, Plio uses Google OAuth Playground Authentication to mimic the exact user behavior on Google Login functionality. Google OAuth credentials will be required to run the test cases that require login.
+
+Follow the links below one-by-one to retrieve required information from Google:
+1. [Google Project and Application Setup](https://docs.cypress.io/guides/testing-strategies/google-authentication#Google-Developer-Console-Setup)
+2. [Using the Google OAuth 2.0 Playground to Create Testing Credentials](https://docs.cypress.io/guides/testing-strategies/google-authentication#Using-the-Google-OAuth-2-0-Playground-to-Create-Testing-Credentials)
+
+### Installation
+1. Install TestCafe for BrowserStack globally:
+    ```sh
+    npm i -g testcafe testcafe-browser-provider-browserstack
+    ```
+
+2. Add the following env variables to your `~/.bash_profile` or `~/.zshrc` file:
+    ```sh
+    export BROWSERSTACK_BASE_URL="http://localhost:8080"
+    export BROWSERSTACK_USERNAME="" # Your BrowserStack username
+    export BROWSERSTACK_ACCESS_KEY="" # Your BrowserStack access key
+    export BROWSERSTACK_PROJECT_NAME="plio"
+    ```
+
+3. Once the env variables are added, restart your terminal or run one of the following command:
+    ```sh
+    source ~/.bash_profile
+    # or
+    source ~/.zshrc
+    ```
+
+### Getting started
+1. To run a test case locally, execute the following command from the project root folder:
+    ```sh
+    # syntax
+    testcafe browser ./path/to/test.spec.js
+
+    # example
+    testcafe chrome ./tests/integration/pages/login.spec.js
```

Comment:
I think adding a line about how to run browserstack tests locally too would be very helpful:

```
testcafe 'browserstack:chrome@96.0:OS X Big Sur','browserstack:firefox@95.0:OS X Big Sur','browserstack:safari@14.1:OS X Big Sur' ./tests/integration --skip-js-errors
```